### PR TITLE
Refactor formatting and rendering. Add color coding and response formatting

### PR DESCRIPTION
**Pull request by lhuett**

### CODE REVIEW

File: insights/formats/text.py [hunk 1]
```diff
@@ -2,7 +2,11 @@
 from pprint import pprint
 from insights import dr, datasource, rule
 from insights.core.context import ExecutionContext
-from insights.formats import Formatter
+from insights.formats import Formatter, render
+from colorama import Fore, Style
+
+
+_counts = {'skip': 0, 'pass': 0, 'rule': 0, 'metadata': 0, 'metadata_key': 0}
```
Comment (csams) [inline insights/formats/text.py]: I'd put this in the class `__init__` and just call it `counts`.

File: insights/formats/text.py [hunk 2]
```diff
@@ -2,7 +2,11 @@
 from pprint import pprint
 from insights import dr, datasource, rule
 from insights.core.context import ExecutionContext
-from insights.formats import Formatter
+from insights.formats import Formatter, render
+from colorama import Fore, Style
```
Comment (csams) [inline insights/formats/text.py]: Maybe we should do something like this for the import so it works even if colorama isn't installed:
```python
from __future__ import print_function                                           
import six                                                                      
try:                                                                            
    from colorama import Fore, Style                                            
except ImportError:                                                             
    print("Install colorama for console colors.")                               
                                                                                
    class Default(type):                                                        
        def __getattr__(*args):                                                 
            return ""                                                           
                                                                                
    class Fore(six.with_metaclass(Default)):                                    
        pass                                                                    
                                                                                
    class Style(six.with_metaclass(Default)):                                   
        pass
```

File: insights/formats/text.py (hunk 1 above)
Comment (lhuett) [reply insights/formats/text.py]: Yep, that makes sense. Then I guess other formatters could use it if they wanted to.

File: insights/formats/text.py (hunk 2 above)
Comment (lhuett) [reply insights/formats/text.py]: Ok, so should the option of installing colorama to get colors be at the users discretion and the colorama dependency in setup.py be removed?

File: insights/formats/__init__.py [hunk 3]
```diff
@@ -7,6 +7,9 @@
 _FORMATTERS = {}
 
 
+counts = {'skip': 0, 'pass': 0, 'rule': 0, 'metadata': 0, 'metadata_key': 0}
```
Comment (csams) [inline insights/formats/__init__.py]: This should go in the formatter class that's using it. Something like
```python
class HumanReadableFormat(Formatter):
...
    def __init__(self, args):
        ...
        self.counts = {'skip': 0, 'pass': 0, 'rule': 0, 'metadata': 0, 'metadata_key': 0}
```

**Pull request merged**